web/event_utils.js

```diff
@@ -138,11 +136,24 @@ class EventBus {
    * @ignore
    */
   _on(eventName, listener, options = null) {
+    let rmAbort = null;
+    if (options?.signal instanceof AbortSignal) {
+      const { signal } = options;
+      if (signal.aborted) {
+        throw new Error("Cannot use an `aborted` signal.");
```

Comment:
If it has been aborted then we're kind of destroying the viewer, right ?
In such a case we could probably just return instead of throwing.
That said it's very likely a corner case pretty hard to reach.